pkg/storage/stores/tsdb/index/index.go

```diff
@@ -1202,19 +1203,18 @@ func NewReader(b ByteSlice) (*Reader, error) {
 	return newReader(b, io.NopCloser(nil))
 }
 
-type nopCloser struct{}
-
-func (nopCloser) Close() error { return nil }
-
 // NewFileReader returns a new index reader against the given index file.
 func NewFileReader(path string) (*Reader, error) {
-	b, err := os.ReadFile(path)
+	f, err := fileutil.OpenMmapFile(path)
```

Comment:
> mmap is generally a PITA for databases, so if we can avoid it that'd be a win IMHO.

Yeah, this is why I initially tried to avoid it, but after running some calculations on larger sized cells, I re-enabled it here in the meantime. I'd love to leave this as a followup with some better investigation when we have a little more time/run into mmap problems in the future!